File: php/features.php
<?php
declare(strict_types=1);

// PollinationsAi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PollinationsAiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PollinationsAiBaseFeature();
            case "test":
                return new PollinationsAiTestFeature();
            default:
                return new PollinationsAiBaseFeature();
        }
    }
}

File: py/pollinationsai_sdk/features.py
# PollinationsAi SDK feature factory

from pollinationsai_sdk.feature.base_feature import PollinationsAiBaseFeature
from pollinationsai_sdk.feature.test_feature import PollinationsAiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PollinationsAiBaseFeature(),
        "test": lambda: PollinationsAiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
